tfjs-backend-webgl/src/webgl_ops_test.ts

```diff
@@ -737,6 +737,90 @@ describeWithFlags('slice and memory usage', WEBGL_ENVS, () => {
     b.dispose();
     expect(getMem().numBytesInGPU).toBe(0);
   });
+
+  it('slice a tensor twice, read it and check memory', async () => {
+    const getMem = () => tf.memory() as WebGLMemoryInfo;
+    expect(getMem().numBytesInGPU).toBe(0);
+
+    // Lazy upload won't increase gpu memory.
+    const a = tf.tensor([2, 3]);
+    expect(getMem().numBytesInGPU).toBe(0);
+
+    // Upload a to the GPU by running an op.
+    a.square().dispose();
+    expect(getMem().numBytesInGPU).toBe(8);
+
+    // Slicing does not allocate new memory.
+    const b = a.slice(0);
+    const c = a.slice(0);
+    expect(getMem().numBytesInGPU).toBe(8);
+
+    // Download a to the CPU but the texture remains on GPU
+    // since b points to it.
+    await a.data();
+    expect(getMem().numBytesInGPU).toBe(8);
+
+    // Dispose a, but the texture should still remain on the GPU
+    // since b points to it.
+    a.dispose();
+    expect(getMem().numBytesInGPU).toBe(8);
+
+    // Dispose b, but the texture should still remain on the GPU
+    // since c points to it.
+    b.dispose();
```

Comment:
Does your fix still work if you call `await b.data(); b.dispose()`?

Reply:
yes